src/target/riscv/riscv-013.c

```diff
@@ -4558,17 +4600,16 @@ static int write_memory_progbuf_run_batch(struct target *target, struct riscv_ba
 
 	if (wait_for_idle(target, &abstractcs) != ERROR_OK)
 		return ERROR_FAIL;
-	RISCV013_INFO(info);
-	info->cmderr = get_field(abstractcs, DM_ABSTRACTCS_CMDERR);
 
+	unsigned int cmderr = get_field(abstractcs, DM_ABSTRACTCS_CMDERR);
```

Comment:
> I guess the cast is optional?

Regarding the cast. There was a discussion in this review to introduce one https://github.com/riscv/riscv-openocd/pull/964#discussion_r1417006866 (since compilers could complain - I tend to agree with this point). Though, likely, it's better to use `get_field32` as you've suggested. Will take a look.